### PR TITLE
TMA-818: executing mufs fails if api returns errors

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/726/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/573/ : FAILURE in 4s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/836/ : SUCCESS in 1m 46s

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 1]
```diff
@@ -415,6 +415,8 @@ def self.execute_mufs(user_filters, options = {})
       end
 
       project_log_formatter.log_user_filter_results(create_results, to_create)
+      create_errors = create_results.select { |r| r[:status] == :failed }
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Calls 'r[:status] == :failed' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Calls 'r[:status]' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Has the variable name 'r' (UncommunicativeVariableName)

File: spec/unit/models/user_filters/user_filter_builder_spec.rb [hunk 2]
```diff
@@ -107,5 +107,16 @@
         end
       end
     end
+
+    context 'when creating MUFs results in errors' do
+      before do
+        allow(client).to receive(:post)
+                           .and_return('userFiltersUpdateResult' => {'failed' => [{status: :failed}]})
```
Comment (yenkins) [inline spec/unit/models/user_filters/user_filter_builder_spec.rb]: Style/SpaceInsideHashLiteralBraces: Space inside { missing.

Comment (yenkins) [inline spec/unit/models/user_filters/user_filter_builder_spec.rb]: Style/SpaceInsideHashLiteralBraces: Space inside } missing.

File: spec/unit/models/user_filters/user_filter_builder_spec.rb [hunk 3]
```diff
@@ -107,5 +107,16 @@
...
+      end
+
+      it 'fails' do
+       expect { subject.execute_mufs(filter_definitions, options) }.to raise_error /Creating MUFs resulted in errors/
```
Comment (yenkins) [inline spec/unit/models/user_filters/user_filter_builder_spec.rb]: Style/IndentationWidth: Use 2 (not 1) spaces for indentation.

Comment (yenkins) [inline spec/unit/models/user_filters/user_filter_builder_spec.rb]: Lint/AmbiguousRegexpLiteral: Ambiguous regexp literal. Parenthesize the method arguments if it's surely a regexp literal, or add a whitespace to the right of the `/` if it should be a division.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/727/ : SUCCESS in 41s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/576/ : FAILURE in 34s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/837/ : SUCCESS in 30s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/728/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/577/ : SUCCESS in 31s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/838/ : SUCCESS in 1m 52s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/729/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/578/ : SUCCESS in 34s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/839/ : SUCCESS in 32s

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 4]
```diff
@@ -445,6 +447,8 @@ def self.execute_mufs(user_filters, options = {})
                        end
 
       project_log_formatter.log_user_filter_results(delete_results, to_delete)
+      delete_errors = delete_results.select { |r| r[:status] == :failed }
+      fail "Creating MUFs resulted in errors: #{delete_errors}" if delete_errors.any?
```
Comment (panjan) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Should it be "Deleting" rather than "Creating"?

Comment (panjan) [reply lib/gooddata/models/user_filters/user_filter_builder.rb]: How are `delete_errors` going to be printed? Can we check them in the spec?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/730/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/579/ : SUCCESS in 36s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/840/ : SUCCESS in 34s

**Comment by panjan** [comment]
LGTM. Thank you very much!

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/268/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/580/ : SUCCESS in 30s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/841/ : SUCCESS in 32s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1192/ : FAILURE in 51m 56s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/950/ : SUCCESS in 39m 25s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/271/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/588/ : SUCCESS in 35s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/849/ : SUCCESS in 37s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1195/ : FAILURE in 58m 35s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/953/ : UNSTABLE in 10m 56s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/273/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/590/ : SUCCESS in 32s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/851/ : SUCCESS in 33s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1197/ : FAILURE in 55m 48s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/955/ : UNSTABLE in 20m 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/736/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/593/ : SUCCESS in 33s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/854/ : SUCCESS in 34s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/276/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/594/ : SUCCESS in 31s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/855/ : SUCCESS in 36s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1200/ : FAILURE in 53m 00s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/958/ : UNSTABLE in 28m 50s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/740/ : SUCCESS in 19s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/602/ : SUCCESS in 32s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/863/ : SUCCESS in 32s

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/294/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/637/ : SUCCESS in 39s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/898/ : SUCCESS in 38s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1224/ : SUCCESS in 58m 40s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/982/ : SUCCESS in 38m 00s

**Pull request merged**